lib/spack/spack/util/module_cmd.py

```diff
@@ -195,9 +195,11 @@ def match_pattern_and_strip(line, pattern, strip=[]):
     def match_flag_and_strip(line, flag, strip=[]):
         flag_idx = line.find(flag)
         if flag_idx >= 0:
-            end = line.find(' ', flag_idx)
-            if end >= 0:
-                path = line[flag_idx + len(flag):end]
+            separators = (' ', '"', "'")
+            occurrences = [line.find(s, flag_idx) for s in separators]
+            indices = [idx for idx in occurrences if idx >= 0]
+            if indices:
+                path = line[flag_idx + len(flag):min(indices)]
```

Comment:
On a second thought, can you add a comment on top of this explaining what's done (i.e. search for the first occurrence of separator after the flag passed as argument etc.)? I think it may help readers in the future. Other than this LGTM.